src/dali/plugin/input/rest/binance_com.py

```diff
@@ -0,0 +1,1011 @@
+                #         "profitAmount": 8.6083060304,   //Earnings Amount
+                #         "coinName":"BTC",              // Coin Type
+                #         "status": 2    //Status：0:Unpaid， 1:Paying  2：Paid
+                #       },
+                #       {
+                #         "time": 1607529600000,
+                #         "coinName": "BTC",
+                #         "type": 0,
+                #         "dayHashRate": 9942053925926,
+                #         "profitAmount": 0.85426469,
+                #         "hashTransfer": 200000000000,
+                #         "transferAmount": 0.02180958,
+                #         "status": 2
+                #       },
+                #       {
+                #         "time": 1607443200000,
+                #         "coinName": "BTC",
+                #         "type": 31,
+                #         "dayHashRate": 200000000000,
+                #         "profitAmount": 0.02905916,
+                #         "hashTransfer": null,
+                #         "transferAmount": null,
+                #         "status": 2
+                #       }
+                #     ],
+                #     "totalNum": 3,          // Total Rows
+                #     "pageSize": 20          // Rows per page
+                #   }
+                # }
+
+                if _DATA in results:
+                    profits: List[Dict[str, Union[int, str]]] = results[_DATA][_ACCOUNTPROFITS]
+                    for result in profits:
+                        self.__logger.debug("Mining profit: %s", json.dumps(result))
+
+                        # Currently the plugin only supports standard mining deposits
+                        # Payment must also be made (status=2) in order to be counted
+                        if result[_TYPE] == 0 and result[_STATUS] == 2:
+                            self._process_gain(result, Keyword.MINING, in_transactions)
+                        else:
+                            self.__logger.error(
+                                "WARNING: Unsupported Mining Transaction Type: %s.\nFull Details: %s\nPlease open an issue at %s.",
+                                result[_TYPE],
+                                json.dumps(result),
+                                self.ISSUES_URL,
+                            )
+
+                    if len(profits) == _MINING_PAGE_LIMIT:
+                        current_page += 1
+                    else:
+                        break
+                else:
+                    break
+
+    def _process_trades(self, in_transactions: List[InTransaction], out_transactions: List[OutTransaction]) -> None:
+
+        ### Regular Trades
+
+        # Binance requires a symbol/market
+        # max limit is 1000
+        for market in self.markets:
+            since: int = self.start_time_ms
+            test: bool = True
+            #       while True:
+            while test:

```

Comment:
I think I need to keep the `while True:` and let the `break` stop the loop if it is under the limit, otherwise move the window forward and retrieve the remaining trades.

Reply:
Sounds good.